### ISSUE TITLE
Custom colors on standard views

### ISSUE THREAD

**Issue by afonso360** [gyscos/Cursive#126]
Hi, would it be possible to implement colors outside of the current theme for some views?

Something similar to this:
```
siv.add_layer(EditView::new()
        .with_color(ColorType::Primary, Color::Rgb(0,0,0))
        .with_id("name"));
```

The backend support for this seems to already exits in `Printer::with_color` it would just be a matter of implementing it in the Trait.

**Comment by gyscos** [comment]
There are two separate issues for this use-case:
* `Printer::with_color` takes a `ColorStyle` (maybe it should be renamed `with_style`...). There's a plan to add a custom `ColorStyle` to use any color: #116 
* `EditView` does not provide any color customization. It currently use the style `Secondary` (in reverse mode), allowing to override this style is a good idea. For now it'll just allow to specify a different style from the theme, but once custom styles are implemented, it'll allow any color configuration.

**Comment by gyscos** [comment]
The second point should be fixed in bdbf6d759e5a5ddfa3353d8313dcbc5d52f801cc:
```rust
siv.add_layer(EditView::new()
        .style(ColorStyle::TitleSecondary)
        .with_id("name"));
```

**Comment by gyscos** [comment]
dad0016759eea654ea2bf3e1166f1da7b572513b brings `ColorStyle::Custom`, enabling out-of-palette colors.
